python/pyspark/pandas/base.py

```diff
@@ -1625,11 +1625,10 @@ def factorize(
         Parameters
         ----------
         sort : bool, default True
```

Comment:
If True, the encoding will be sorted, otherwise the order of encoding depends on the order of appearance of the values. 
it that right?

Reply:
This is what pandas have in 2.0.3 
Sort uniques and shuffle codes to maintain the relationship.
https://pandas.pydata.org/docs/reference/api/pandas.factorize.html

Reply:
> If True, the encoding will be sorted, otherwise the order of encoding depends on the order of appearance of the values.

Yes, but seems like we currently don't support `False` for `sort` for now in pandas API on Spark for some reason. @xinrong-meng do you remember why we only support only `True` for `sort` parameter?? (as an original author for this functions from [old PR](https://github.com/databricks/koalas/pull/1972) 😂 )